service/src/worker.rs

```diff
@@ -84,35 +75,38 @@ where
 		}
 		Ok(())
 	}
+}
 
-	fn peers(&self) -> WorkerResult<Vec<EnclaveMetadata>> {
-		Ok(self.node_api.all_enclaves(None)?)
+/// Looks for new peers and updates them.
+pub trait UpdatePeers {
+	fn search_peers(&self) -> WorkerResult<Vec<Url>>;
+	fn store_peers(&mut self, peers: Vec<Url>) -> WorkerResult<()>;
+	fn update_peers(&mut self) -> WorkerResult<()> {
+		let peers = self.search_peers()?;
+		self.store_peers(peers)
 	}
 }
 
-/// Temporary method that transforms the workers rpc port of the direct api defined in rpc/direct_client
-/// to the new version in rpc/server. Remove this, when all the methods have been migrated to the new one
-/// in rpc/server.
-pub fn worker_url_into_async_rpc_url(url: &str) -> WorkerResult<String> {
```

Comment:
Is obsolete now, as we give the untrusted worker url as a parameter at worker start up.